contracts/eternal-storage/EternalStorage.sol

```diff
@@ -14,21 +14,25 @@ contract EternalStorage {
     // Address of the current implementation
     address internal _implementation;
 
+    // Address of the owner of the contract
+    address internal _owner;
+
+    /**
+     * @dev Access check: ensure that either we are on the genesis block, or
+     * `msg.sender` is the owner of the contract.  The genesis block is
+     * hard-coded into the client, so attacks based on it are not possible.
+     */
+    modifier onlyOwner() {
+        require(msg.sender == _owner || block.number == 0);
```

Comment:
What is `block.number == 0` here for?

Reply:
It was from when I was trying to get the smart contracts to work at all, and thought that Parity initialized the smart contracts on the genesis block in an unspecified order.  Since `msg.sender` is `address(0)` on the genesis block, this is not necessary.